plugins/destination/s3/client/spec.go

```diff
@@ -0,0 +1,32 @@
+package client
+
+import (
+	"fmt"
+)
+
+type FormatType string
+
+const (
+	FormatTypeCSV  = "csv"
+	FormatTypeJSON = "json"
+)
+
+type Spec struct {
+	Bucket   string     `json:"bucket,omitempty"`
+	Key      string     `json:"key,omitempty"`
+	Format   FormatType `json:"format,omitempty"`
+	NoRotate bool       `json:"no_rotate,omitempty"`
+}
+
+func (*Spec) SetDefaults() {}
+
+func (s *Spec) Validate() error {
```

Comment:
Should we validate `Key`? The docs say it's required

Reply:
Yeah, Im also going to change it to `Path` to be consistent with all others destinations

Reply:
Why not call it `prefix`?